src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -48,28 +51,77 @@ def _convert_v2_artifact_to_v1_dict(
     return v2_artifact
 
 
-def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
+def _pdf_v2_to_v1(
+    v2_definition: DefinitionV20,
+    package_entity_id: str = "",
+    app_entity_id: str = "",
+) -> DefinitionV11:
     pdfv1: Dict[str, Any] = {"definition_version": "1.1", "native_app": {}}
 
     app_package_definition: Optional[ApplicationPackageEntityModel] = None
     app_definition: Optional[ApplicationEntityModel] = None
 
+    # Enumerate all application package and application entities in the project definition
+    packages = {}
+    apps = {}
```

Comment:
You can use `get_entities_by_type()` instead of the iteration below:
```suggestion
    packages: Dict[str, ApplicationPackageEntityModel] = v2_definition .get_entities_by_type("application package")
    apps: Dict[str, ApplicationEntityModel] = v2_definition .get_entities_by_type("application")
```

Reply:
updated